src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -581,6 +581,41 @@ public async Task GoAwayFrame_NewRequest_NewConnection()
             }
         }
 
+        [ConditionalFact(nameof(SupportsAlpn))]
+        public async Task GoAwayFrame_ServerDisconnect_AbortStreams()
+        {
+            using (var server = Http2LoopbackServer.CreateServer())
+            using (HttpClient client = CreateHttpClient())
+            {
+                server.AllowMultipleConnections = true;
+
+                Task<HttpResponseMessage> sendTask = client.GetAsync(server.Address);
+                Http2LoopbackConnection connection = await server.EstablishConnectionAsync();
+                int streamId = await connection.ReadRequestHeaderAsync();
+
+                await connection.SendGoAway(streamId);
+
+                // wait until GOAWAY received
+                await connection.PingPong();
+                await connection.WaitForConnectionShutdownAsync();
+
+                // Timeout is small on purpose, connection should be closed immediatelly
+                // We add some slack though in case machine is being slow
+                const int ShortTimeout = 2000;
+                const int ErrorMargin = 500;
+
+                Stopwatch sw = Stopwatch.StartNew();
+                await Assert.ThrowsAnyAsync<HttpRequestException>(() =>
+                    new Task[]
+                    {
+                        sendTask
+                    }.WhenAllOrAnyFailed(ShortTimeout));
+
+                // In case we timeout for longer but exception matches
+                Assert.InRange(sw.Elapsed, TimeSpan.Zero, TimeSpan.FromMilliseconds(ShortTimeout + ErrorMargin));
```

Comment:
I don't understand the purpose of this line.  Seems like it can just be deleted.

Reply:
I added it in case we somehow got correct exception type but it actually timed out. WIll remove this and increase timeouts as per previous comment and rely only on exception type